File: problems/coding/013-container-most-water/solution.py
def maxArea(h):
    i, j, best = 0, len(h) - 1, 0
    while i < j:
        best = max(best, (j - i) * min(h[i], h[j]))
        if h[i] < h[j]: i += 1
        else: j -= 1
    return best

File: problems/coding/013-container-most-water/solution.js
function maxArea(h) {
  let i = 0, j = h.length - 1, best = 0;
  while (i < j) { best = Math.max(best, (j - i) * Math.min(h[i], h[j])); if (h[i] < h[j]) i++; else j--; }
  return best;
}
